operator/builtin/input/file/file_test.go

```diff
@@ -469,24 +470,6 @@ func TestFileBatching(t *testing.T) {
 	}
 
 	require.ElementsMatch(t, expectedMessages, actualMessages)
-
-	// Write more logs to each file so we can validate that all files are still known
-	for i, temp := range temps {
-		for j := 0; j < linesPerFile; j++ {
-			message := fmt.Sprintf("%s %d %d", stringWithLength(20), i, j)
-			temp.WriteString(message + "\n")
-			expectedMessages = append(expectedMessages, message)
-		}
-	}
-
-	for b := 0; b < expectedBatches; b++ {
-		// poll once so we can validate that files were batched
-		operator.poll(context.Background())
-		actualMessages = append(actualMessages, waitForN(t, logReceived, expectedLinesPerBatch)...)
-		expectNoMessagesUntil(t, logReceived, 10*time.Millisecond)
-	}
-
-	require.ElementsMatch(t, expectedMessages, actualMessages)
```

Comment:
I think these changes have invalidated the test.

Tinkering with the removed code shows me that the first `for` loop does not cause an issue. In other words, we can still "Write more logs to each file ...".

However, any additional call to `operator.poll` will never return.

I think this needs to be understood carefully, and that it likely indicates a problem with this implementation.

Reply:
It was because the mock channel created has limited size. I doubled it and it works. at the first poll cycle of 2nd for loop, it tails 20 files that have data added. so the number of events in that cycle is double of rest of the cycles. Anyway, The MaxConcurrentFiles is respected.